mod/01_rating/set/type/metric_value.rb

```diff
@@ -149,36 +149,138 @@ def fill_subcards metric_value, source_card
 end
 
 format :html do
-
   view :new do |args|
     # return super(args)
-    if Env.params[:noframe]
-      form_opts = args[:form_opts] ? args.delete(:form_opts) : {}
-      form_opts[:hidden] = args.delete(:hidden)
-      form_opts['main-success'] = 'REDIRECT'
-      card_form :create, form_opts do
-        output [
-          _optional_render(:name_formgroup, args),
-          _optional_render(:type_formgroup, args),
-          _optional_render(:content_formgroup, args),
-          _optional_render(:button_formgroup, args)
-        ]
-      end
+    return _render_no_frame_form args if Env.params[:noframe]
+    # return super(args) if args[:source] || args[:metric] || args[:company]
+    return super(args) if args[:source] || args[:company]
+    @form_root = true
+
+    frame args do # no form!
+      [
+        _optional_render(:content_formgroup,
+                         args.merge(metric_value_landing: true))
+        #  _optional_render(:button_formgroup, button_args)
+      ]
+    end
+  end
+
+  view :editor do |args|
+    if args[:company] && args[:metric]
+      _render_metric_company_add_value_editor args
+    # elsif args[:source] || args[:metric]
+    #   _render_add_value_editor args
+    elsif args[:metric_value_landing]
+      _render_metric_value_landing args
     else
-      super(args)
+      super(args.merge(edit_fields: '+value'))
     end
   end
 
+  view :metric_value_landing do |args|
+    metric_field = _render_metric_field(args)
+    render_haml source_container: _render_source_container,
+                metric_field: metric_field do
+      <<-HAML
+.col-md-6.border-right.panel-default
+  %h4
+    Company
+  %hr
+    = field_nest :wikirate_company, title: ''
+  %h4
+    Metric
+  %hr
+    = metric_field
+= source_container
+      HAML
+    end
+  end
 
-  def get_structure args
-    if args[:company]
-      'metric company add value'
-    elsif args[:source]
-      'metric_source_add_value'
-    elsif args[:metric]
-      'metric add value'
-    else
-      'default add metric value'
+  view :metric_field do |args|
+    metric = args[:metric]
+    metric_field =
+      Card.fetch(card.cardname.field(:metric), new: { content: metric })
+    render_haml metric: metric,
+                source_container: _render_source_container,
+                metric_field: metric_field do
+      <<-HAML
+- if metric
+  = nest metric_field, title: ''
+- else
+  = field_nest :metric, title: ''
```

Comment:
I don't why you need the second case. Doesn't `nest metric_field, title: ''` do the same if `metric` is empty?